File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSecondaryFixedDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryFixedDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd31
            AppTheme.STANDARD -> R.color.cstd31
            AppTheme.BLUE -> R.color.cbd31
            AppTheme.GREEN -> R.color.cgd31
            AppTheme.RED -> R.color.crd31
            AppTheme.NOTHING -> R.color.cnod31
        }
    }
}

File: lib/layer2_core/dark/e_on_secondary_fixed_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSecondaryFixedDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd31;
      case 'standard': return cstd31;
      case 'blue': return cbd31;
      case 'green': return cgd31;
      case 'red': return crd31;
      case 'nothing': return cnod31;
      default: return cstd31;
    }
  }
}
